src/integration-test/java/com/commercetools/sync/integration/services/impl/ShoppingListServiceImplIT.java

```diff
@@ -0,0 +1,265 @@
+        assertThat(cache).hasSize(1);
+
+        cache = shoppingListService.cacheKeysToIds(singleton(shoppingList.getKey()))
+                                   .toCompletableFuture().join();
+        assertThat(cache).hasSize(1);
+
+        verify(spyClient, times(1)).execute(any());
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+    }
+
+    @Test
+    void createShoppingList_WithValidShoppingList_ShouldCreateShoppingList() {
+        //preparation
+        ProductType productType = createProductType(PRODUCT_TYPE_RESOURCE_PATH, CTP_TARGET_CLIENT);
+        final ProductDraft productDraft = ProductDraftBuilder
+            .of(ResourceIdentifier.ofKey(productType.getKey()), LocalizedString.ofEnglish("newProduct"),
+                LocalizedString.ofEnglish("foo"),
+                ProductVariantDraftBuilder.of().key("foo-new").sku("sku-new").build())
+            .key("newProduct")
+            .build();
+        executeBlocking(CTP_TARGET_CLIENT.execute(ProductCreateCommand.of(productDraft)));
+        LineItemDraft lineItemDraft = LineItemDraftBuilder.ofSku("sku-new", Long.valueOf(1)).build();
+        final ShoppingListDraftDsl newShoppingListDraft =
+            ShoppingListDraftBuilder.of(LocalizedString.ofEnglish("new_name"))
+                                    .key("new_key")
+                                    .plusLineItems(lineItemDraft)
+                                    .build();
+
+        final SphereClient spyClient = spy(CTP_TARGET_CLIENT);
+
+        final ShoppingListSyncOptions options =
+            ShoppingListSyncOptionsBuilder.of(spyClient)
+                                          .errorCallback((exception, oldResource, newResource, actions) -> {
+                                              errorCallBackMessages.add(exception.getMessage());
+                                              errorCallBackExceptions.add(exception);
+                                          })
+                                          .build();
+        ShoppingListService spyShoppingListService = new ShoppingListServiceImpl(options);
+
+        //test
+        final Optional<ShoppingList> createdShoppingList =
+            spyShoppingListService.createShoppingList(newShoppingListDraft).toCompletableFuture().join();
+
+        final Optional<ShoppingList> queriedOptional = CTP_TARGET_CLIENT
+            .execute(ShoppingListQuery.of().withPredicates(shoppingListQueryModel ->
+                shoppingListQueryModel.key().is("new_key")))
+            .toCompletableFuture().join().head();
+
+        assertThat(queriedOptional).hasValueSatisfying(queried ->
+            assertThat(createdShoppingList).hasValueSatisfying(created -> {
+                assertThat(created.getKey()).isEqualTo(queried.getKey());
+                assertThat(created.getName()).isEqualTo(queried.getName());
+                assertThat(created.getLineItems()).hasSize(1);
+            }));
+
```

Comment:
empty spaces like this might be removed for better readability.

Reply:
2365da1